.github/actions/spell-check/whitelist.txt

```diff
@@ -0,0 +1,3240 @@
+conattrs
+conbufferout
+concat
+concfg
+conddkrefs
+condrv
+conechokey
+conemu
+config
+configurability
+conhost
+conhostv
+conime
+conimeinfo
+conint
+conintegrity
+conintegritylib
+conintegrityuwp
+coninteractivitybase
+coninteractivityonecore
+coninteractivitywin
+conio
+coniosrv
+CONKBD
+conlibk
+conmsgl
+CONNECTINFO
+CONOUT
+conpixels
+conprops
+conpropsp
+conpty
+conptylib
+consecteturadipiscingelit
+conserv
+Consolas
+consoleaccessibility
+consoleapi
+CONSOLECONTROL
+CONSOLEENDTASK
+CONSOLEIME
+consoleinternal
+Consoleroot
+Consolescreen
+CONSOLESETFOREGROUND
+consoletaeftemplates
+CONSOLEV
+Consolewait
+CONSOLEWINDOWOWNER
+consrv
+constexpr
+constexprable
+constness
+contentfiles
+conterm
+CONTEXTMENU
+controlkeystates
+contsf
+contypes
+convarea
+conwinuserrefs
+coord
+coordnew
+COPYCOLOR
+CORESYSTEM
+cotaskmem
+countof
+cout
+CParams
+CPG
+cpinfo
+CPINFOEX
+cplinfo
+cplusplus
+cpp
+cppcorecheck
+cppcorecheckrules
+cpprest
+cpprestsdk
+cppwinrt
+CProc
+cpx
+crbegin
+CREATESCREENBUFFER
+CREATESTRUCT
+CREATESTRUCTW
+creativecommons
+cred
+cref
+crend
+CRLFs
+crloew
+crt
+CRTLIBS
+csbi
+csbiex
+csdef
+csharp
+CSHORT
+cso
+csproj
+Csr
+csrmsg
+CSRSS
+csrutil
+cstdio
+cstr
+cstring
+cstyle
+CSwitch
+csx
+CText
+ctl
+ctlseqs
+Ctlv
+ctor
+CTRLEVENT
+CTRLKEYSHORTCUTS
+Ctx
+Ctxt
+ctype
+CUF
+cupxy
+curated
+CURRENTFONT
+currentmode
+CURRENTPAGE
+CURSORCOLOR
+CURSORSIZE
+CURSORTYPE
+customframe
+CUU
+cvd
+cw
+Cwa
+cwch
+cwchar
+cwctype
+cwd
+cx
+CXFRAME
+CXFULLSCREEN
+CXHSCROLL
+CXMIN
+CXPADDEDBORDER
+CXSIZE
+CXSMICON
+CXVIRTUALSCREEN
+CXVSCROLL
+cxx
+CYFRAME
+CYFULLSCREEN
+cygwin
+CYHSCROLL
+CYMIN
+CYPADDEDBORDER
+CYSIZE
+CYSIZEFRAME
+CYSMICON
+CYVIRTUALSCREEN
+CYVSCROLL
+DADADA
+dahall
+dai
+DATABLOCK
+datacollector
+dataprotection
+dataprotectionprovider
+DATAVIEW
+DATAW
+Datei
+Dateien
+dba
+DBatch
+dbb
+dbcs
+DBCSCHAR
+DBCSFONT
+dbd
+DBDCF
+DBDDE
+dbf
+DBFFu
+dbg
+DBGALL
+DBGCHARS
+DBGFONTS
+DBGOUTPUT
+dbh
+dblclk
+DBlob
+dbmdl
+dbproj
+DBUILD
+DCDFE
+dcf
+DColor
+DCompile
+dcompiler
+ddb
+dde
+DDESHARE
+DDevice
+DEADCHAR
+dealloc
+debian
+debolden
+debounce
+DECALN
+DECANM
+DECAUPSS
+DECAWM
+DECCKM
+DECCOLM
+decf
+DECKPAM
+DECKPM
+DECKPNM
+DECLL
+DECLRMM
+decls
+declspec
+decltype
+DECNKM
+DECNRCM
+DECOM
+deconstructed
+DECPCTERM
+DECRC
+DECRLM
+DECRST
+DECSASD
+DECSC
+DECSCA
+DECSCNM
+DECSCPP
+DECSCUSR
+DECSED
+DECSEL
+DECSET
+DECSLRM
+DECSMBV
+DECSR
+decstandar
+DECSTBM
+DECSTR
+DECTCEM
+Dedupe
+deduplicated
+DEFAULTBACKGROUND
+DEFAULTFOREGROUND
+defaultmk
+defaultsettings
+DEFAULTTONEAREST
+DEFAULTTONULL
+DEFAULTTOPRIMARY
+defectdefs
+DEFERERASE
+deff
+DEFFACE
+defing
+DEFPUSHBUTTON
+DELAYLOAD
+deletable
+DELETEONRELEASE
+delims
+Delt
+demoable
+depersist
+deprioritized
+deque
+deref
+deserialization
+deserialize
+deserialized
+deserializer
+deserializing
+desktopwindowxamlsource
+dest
+DESTINATIONNAME
+devblogs
+developercommunity
+devicecode
+devicefamily
+devops
+Dext
+df
+dfa
+DFactory
+DFF
+DFFCAC
+DFFF
+DFFFu
+DFu
+DGCZZu
```

Comment:
Where is stuff like `DGCZZu` coming from? That feels like nonsense that should be put in its own bucket and not applied universally.

Reply:
hm, base64?

Reply:
YouTube:
https://github.com/microsoft/terminal/blob/941a44a46491e1a018419eb24c9d92d3b2640e55/README.md#L112
https://github.com/microsoft/terminal/blob/c879cd3f9722bec677ff10f8f4452a3c3c0d5cff/doc/terminal-v1-roadmap.md#L25

Reply:
Wow that's curious. If we had regex, I'd be inclined to make it ignore all of a URL pattern for regexs.

Reply:
I've found plenty of typos in URLs... Some can be fixed, e.g. some places include a numerical ID which means one can change the human readable portion :-), and others can be fixed by e.g. updating the website content...

That said, being able to exclude things like youtube urls does seem useful... I'll have to think about how to deal w/ that.

My imaginary pipeline is something like:
1. select files -- **excludes.txt**
2. filter input -- my spelling repo has [code for things like this](https://github.com/jsoref/spelling/blob/a50c32659d3f93117a5edc90d8c7b0ba0d6919a5/hgrmjunk#L63-L66) -- specifically to address `data:` URLs (a key source of base64) and general hex tokens -- but I haven't thought through how to do it here. As you can see, I'm using regular expressions, so yes, that's probably what I'd go w/. Again, probably one regexp per line.
3. [parse corpus](https://github.com/check-spelling/check-spelling/blob/master/spelling-unknown-word-splitter.pl)
4. drop words found in dictionary -- currently the dictionary isn't configurable in the action (I intend to support this), and the proposal to add things like curated apis would plug in here
5. filter output -- **whitelist.txt** -- this phase is designed to warn about stale whitelist entries (and includes code to automatically remove entries from it as well as adding new entries).

Would you want multiple whitelist files, or would it be sufficient to have multiple "dictionary" files? -- The distinction is that dictionaries are treated as read-only -- the tool will not warn you about unused items in them. Whereas the tool will complain if you've whitelisted something and it can't find it in the whitelist.

It's easier for me to imagine code that handles multiple dictionaries and a single whitelist (especially if I add the code to handle instance counts). Although if I have to handle comments, then I can probably make things work using some fancy greps.
-- I think I might also drop support for having the whitelists live outside of the current branch, right now I'm still theoretically supporting an older model where the tool retrieves the whitelist+excludes files from remote locations, but that won't work well w/ enumerating files...

Reply:
I've finally implemented regexp support along w/ multiple file support and dictionaries. I've just published it and am letting it chew on this repository....

The lazy approach to generating a dictionary would be to create a repository with just the spellchecker and all the files whose contents you believe are ok, the warning output should be valid input as a dictionary file (instead of as a whitelist file). -- I'll have to document this approach.

If that's too inconvenient, I can think about other ways of handling the problem.

While you might be comfortable checking in a script into your repository, I worry that others might not.